src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldEntitySpawner.java

```diff
@@ -208,7 +208,7 @@ public int findChunksForSpawning(WorldServer worldServerIn, boolean spawnHostile
                         int k1 = blockpos.getX();
                         int l1 = blockpos.getY();
                         int i2 = blockpos.getZ();
-                        IBlockState iblockstate = worldServerIn.getBlockState(blockpos);
+                        IBlockState iblockstate = chunk.getBlockState(blockpos);
```

Comment:
Why bypass the buildheight checks? You could be getting block states that are outside the build height, which vanilla will technically attempt to perform.

Reply:
Doesn't `getRandomChunkPosition` return a valid blockpos ?